.github/workflows/ngen_integration.yaml

```diff
@@ -0,0 +1,111 @@
+  # TODO: add something later that can check the runners for hyperthreading
+  LINUX_NUM_PROC_CORES: 2
+  MACOS_NUM_PROC_CORES: 3
+  ASAN_OPTIONS: detect_leaks=false
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # Run general unit tests in linux environment
+  test_surfacebmi_plus_cfe:
+    # The type of runner that the job will run on
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checkout and build the evapotranspiration
+      - name: Checkout the commit
+        uses: actions/checkout@v4
+
+      - name: Save Evapotranspiration to a Temp Directory
+        run: |
+          # Move files to appropriate temporary directory
+          mkdir ${{runner.temp}}/evapotranspiration
+          mv ./* ${{runner.temp}}/evapotranspiration
+
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: checkout actions in another repo
+        uses: actions/checkout@v4
+        with:
+          repository: noaa-owp/ngen
+
+      - name: Build Surfacebmi
+        id: submod_build_1
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: " extern/noah-owp-modular/"
+          targets: "surfacebmi"
+
+      - name: Build ISO C Fortran BMI
+        id: submod_build_2
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/iso_c_fortran_bmi/"
+
+      - name: Build CFE
+        id: submod_build_3
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/cfe/"
+          targets: "cfebmi"
+
+      - name: Build SLoTH
+        id: submod_build_6
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/sloth/"
+          targets: "slothmodel"
+
+      # Build Ngen and save some files
+      - name: Build Ngen
+        id: ngen_id1
+        uses: ./.github/actions/ngen-build
+        with:
+          targets: "ngen"
+          build-cores: ${{ env.LINUX_NUM_PROC_CORES }}
+          #is this required for this test?
+          bmi_c: 'ON'
+          bmi_fortran: 'ON'
+        timeout-minutes: 15
+
+      - name: Prepare to Build Evapotranspiration Library for Ngen
+        run: |
+          # Move files to appropriate directory to build evapotranspiration library
+          rm -rf extern/evapotranspiration/evapotranspiration/*
+          mv ${{runner.temp}}/evapotranspiration/* extern/evapotranspiration/evapotranspiration
+
+      - name: Build PET
+        id: submod_build_5
+        uses: ./.github/actions/ngen-submod-build
+        with:
+          mod-dir: "extern/evapotranspiration/evapotranspiration"
+          targets: "petbmi"
+
+      - name: Run Ngen Test
```

Comment:
Unclear why this is here, "Run ngen with topmodel with CFE"?  The [`example_bmi_multi_realization_config.json`](https://github.com/NOAA-OWP/ngen/blob/master/data/example_bmi_multi_realization_config.json) couples cfe-noahowp. Recommend removing.

Reply:
This is my oversight. I should have used a realization file in the test that uses the PET. Will change that,

Reply:
Looks like I did have a second test that uses PET. I can remove the one that does not use PET, but it probably good to have it there.

Reply:
I would just remove it for the time being.  The topmod library isn't even being built in this workflow.  The `Run surfacebmi, cfebmi and petbmi` should suffice.  We can always add more builds later on. -thanks